pkg/deployment/note/show_test.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+	multierror "github.com/hashicorp/go-multierror"
+
+	"github.com/elastic/ecctl/pkg/deployment"
+	"github.com/elastic/ecctl/pkg/util"
+)
+
+func TestGet(t *testing.T) {
+	var messageDateTime = util.ParseDate(t, "2018-04-13T07:11:54.999Z")
+	const simpleNote = `{
+		"id": "1",
+		"message": "a message",
+		"user_id": "root",
+		"timestamp": "2018-04-13T07:11:54.999Z"
+	}`
+	const getResponse = `{
+  "healthy": true,
+  "id": "e3dac8bf3dc64c528c295a94d0f19a77",
+  "resources": {
+    "elasticsearch": [{
+      "id": "418017cd1c7f402cbb7a981b2004ceeb",
+      "ref_id": "main-elasticsearch",
+      "region": "ece-region"
+    }]
+  }
+}`
+
+	type args struct {
+		params GetParams
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *models.Note
+		wantErr error
+	}{
+		{
+			name: "Get note succeeds",
+			args: args{params: GetParams{Params: Params{
+				NoteID: "1",
+				Params: deployment.Params{
+					ID: "e3dac8bf3dc64c528c295a94d0f19a77",
+					API: api.NewMock(mock.Response{Response: http.Response{
+						Body:       mock.NewStringBody(getResponse),
+						StatusCode: 200,
+					}},
+						mock.Response{Response: http.Response{
+							Body:       mock.NewStringBody(simpleNote),
+							StatusCode: 200,
+						}}),
+				},
+			}}},
+			want: &models.Note{
+				ID:        "1",
+				Message:   ec.String("a message"),
+				UserID:    "root",
+				Timestamp: messageDateTime,
+			},
+		},
+		{
+			name: "Get note fails due to api error (fails to get deployment)",
+			args: args{
+				params: GetParams{Params: Params{
+					NoteID: "1",
+					Params: deployment.Params{
+						ID: "a2c4f423c1014941b75a48292264dd25",
+						API: api.NewMock(mock.Response{
+							Response: http.Response{
+								StatusCode: http.StatusNotFound,
+								Status:     http.StatusText(http.StatusNotFound),
+								Body:       mock.NewStringBody(`{}`),
+							},
+						}),
+					},
+				}},
+			},
+			wantErr: errors.New(errNull),
+		},
+		{
+			name: "Get note fails due to api error",
+			args: args{
+				params: GetParams{Params: Params{
+					NoteID: "1",
+					Params: deployment.Params{
+						ID: "a2c4f423c1014941b75a48292264dd25",
+						API: api.NewMock(mock.Response{Response: http.Response{
+							Body:       mock.NewStringBody(getResponse),
+							StatusCode: 200,
+						}},
+							mock.Response{
+								Response: http.Response{
+									StatusCode: http.StatusNotFound,
+									Status:     http.StatusText(http.StatusNotFound),
+									Body:       mock.NewStringBody(`{}`),
+								},
+							}),
+					},
+				}},
+			},
+			wantErr: errors.New(errNull),
+		},
+		{
+			name: "Get note fails due to validation",
+			args: args{
+				params: GetParams{
+					Params: Params{
+						Params: deployment.Params{},
+					},
+				},
+			},
+			wantErr: &multierror.Error{Errors: []error{
+				errors.New("note id cannot be empty"),
+				util.ErrAPIReq,
```

Comment:
I'd use the plan text here instead of the utility constant. The reason is that if accidentally we change the `ErrAPIReq` value to something that doesn't make sense the test will keep succeeding while it shouldn't 😄